test/functional/tests/io_class/test_io_class_stats_file_size_core.py

```diff
@@ -0,0 +1,234 @@
+#
+
+from itertools import cycle
+
+import pytest
+
+from datetime import timedelta
+from api.cas import casadm, ioclass_config
+from api.cas.cache_config import CacheMode, CleaningPolicy, SeqCutOffPolicy
+from api.cas.ioclass_config import IoClass
+from core.test_run_utils import TestRun
+from storage_devices.disk import DiskTypeSet, DiskType, DiskTypeLowerThan
+from test_tools import fs_utils
+from test_tools.disk_utils import Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_utils.os_utils import sync, drop_caches
+from test_utils.size import Size, Unit
+
+
+io_class_sizes = [4096, 16384, 65536, 262144, 1048576, 4194304,
+                  16777216, 67108864, 268435456, 1073741824]
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_stats_file_size_core():
+    """
+    title: Open CAS statistics values for file size - per core.
+    description: Check Open CAS ability to assign correct IO class to file size.
+    pass_criteria:
+        - after FIO without direct statistics increases only for tested IO class
+    """
+
+    with TestRun.step("Prepare devices."):
+        cache_devices, core_devices = prepare_devices()
+
+    with TestRun.step("Start caches (one for each supported cache mode) and add core devices."):
+        caches, cores = prepare_caches_and_cores(cache_devices, core_devices)
+
+    with TestRun.step("Make filesystem on OpenCAS devices and mount it."):
+        for core, fs in zip(cores, cycle(Filesystem)):
+            mount_point = core.path.replace('/dev/', '/mnt/')
+            core.create_filesystem(fs)
+            core.mount(mount_point)
+            sync()
+
+    with TestRun.step("In default IO class configuration file set Eviction priority=1 "
+                      "and load it for all caches."):
+        io_classes = prepare_io_classes(caches, io_class_sizes)
+
+    with TestRun.step("Run fio for all devices checking IO class statistics."):
+        sizes = prepare_fio_sizes(io_class_sizes)
+        size_min = Size(512, Unit.Byte)
+
+        for io_class, size in zip(io_classes[:-1], sizes):
+            with TestRun.step(f"Testing IO class {io_class.id}."):
+                with TestRun.step("Run fio for each device and check that "
+                                  "only for tested IO class statistics increases."):
+                    for core in cores:
+                        core.reset_counters()
+                        fio = fio_params(core, size_min, size)
+                        result = fio.run()
+                        sync()
+                        drop_caches()
+                        issued_reqs_no = \
+                            result[0].write_requests_number() + result[0].read_requests_number()
+                        check_statistics(caches[core.cache_id-1], core, io_classes[:-1],
```

Comment:
`caches[core.cache_id-1]` -> `cache`

Reply:
`io_classes[:-1]` -> `io_classes` because it might be worthwhile to check direct ioclass stats as well